tests/account_tests.cpp

```diff
@@ -0,0 +1,134 @@
+#include "token_tester.hpp"
+
+const static name accounts_name = N(gxc.account);
+
+class gxc_account_tester : public gxc_token_tester {
+public:
+
+   gxc_account_tester() {
+      create_accounts({ accounts_name });
+      produce_blocks(1);
+
+      _set_code(accounts_name, contracts::account_wasm());
+      _set_abi(accounts_name, contracts::account_abi().data());
+      produce_blocks(1);
+
+      auto accnt = control->db().get<account_object,by_name>(accounts_name);
+      abi_def abi;
+      BOOST_REQUIRE_EQUAL(abi_serializer::to_abi(accnt.abi, abi), true);
+      abi_ser[accounts_name].set_abi(abi, abi_serializer_max_time);
+   }
+
+   action_result setnick(account_name name, string nickname) {
+      return PUSH_ACTION(accounts_name, name, (name)(nickname));
+   }
+
+   action_result setpartner(account_name name, bool is_partner = true) {
+      return PUSH_ACTION(accounts_name, name, (name)(is_partner));
+   }
+
+   action_result connect(name name, struct name service, string auth_token) {
```

Comment:
This is not mandatory, but you can change parameter name from `name name` to `name _name` to avoid using `sturct name` to declare `eosio::name` type in following lines.
In contract, parameter name determines the key name in ABI, so its name is important, but not same here. It would be enough if you can send correctly serialized action.

Reply:
thank you for your teaching